addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/EnvFileScanRuleUnitTest.java

```diff
@@ -117,4 +145,57 @@ void shouldNotAlertWhenRequestIsSuccessfulButContentNotRelevant() throws Excepti
         // Then
         assertThat(alertsRaised, hasSize(0));
     }
+
+    @Test
+    void shouldNotAlertWhenRequestIsSuccessfulButContentIsRelevantWrongType() throws Exception {
+        // Given
+        String path = "/.env";
+        this.nano.addHandler(
+                createHandler(
+                        path,
+                        newFixedLengthResponse(
+                                Response.Status.OK, NanoHTTPD.MIME_HTML, RELEVANT_BODY)));
+        HttpMessage msg = this.getHttpMessage(path);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    void shouldNotAlertWhenResponseIsNotOK() throws Exception {
+        // Given
+        String path = "/.env";
+        this.nano.addHandler(
+                createHandler(
+                        path,
+                        newFixedLengthResponse(
+                                Response.Status.BAD_REQUEST,
+                                NanoHTTPD.MIME_HTML,
+                                "Application Error")));
```

Comment:
The preconditions are not correct, the Content-Type/body used would prevent the rule from raising an alert (even if the status code is wrong/removed the test still passes).

Reply:
fixed